### PR TITLE
Dbaas user managed sql mode

### PR DESCRIPTION
**Pull request by gregmankes**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/cpuguy83/go-md2man v1.0.10 // indirect
 	github.com/creack/pty v1.1.7
-	github.com/digitalocean/godo v1.22.0
+	github.com/digitalocean/godo v1.26.0
```
Comment (bentranter) [inline go.mod]: This is done by Kamal's currently open PR, so once he merges that, you're probably good to rebase against that to get `godo v1.26.0`.

File: integration/database_sql_mode_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+				reqBody, err := ioutil.ReadAll(req.Body)
+				expect.NoError(err)
+
+				request := struct {
+					SQLMode string `json:"sql_mode"`
+				}{}
+
+				err = json.Unmarshal(reqBody, &request)
+				expect.NoError(err)
```
Comment (bentranter) [inline integration/database_sql_mode_test.go]: To make this consistent with the other tests, could you write the JSON body as a raw string, and compare the marshalled JSON against that as an assertion? Something like,

https://github.com/digitalocean/doctl/blob/master/integration/firewall_create_test.go#L27-L44

File: commands/databases.go [hunk 3]
```diff
@@ -909,3 +910,54 @@ func displayDatabaseReplicaConnection(c *CmdConfig, conn do.DatabaseConnection)
...
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:     "sql-mode",
+			Aliases: []string{"sm"},
+			Short:   "database sql-mode commands",
+			Long:    "sql-mode is used to access database sql-mode commands",
+		},
+	}
+
+	CmdBuilder(cmd, RunDatabaseGetSQLModes, "get <database-id>",
```
Comment (zachgersh) [inline commands/databases.go]: get should have a `g` alias for the command.

File: commands/databases_test.go [hunk 4]
```diff
@@ -858,3 +874,54 @@ func TestDatabasesReplicaDelete(t *testing.T) {
 		assert.EqualError(t, err, errTest.Error())
 	})
 }
+
+func TestDatabaseGetSQLModes(t *testing.T) {
+	// Successful
```
Comment (zachgersh) [inline commands/databases_test.go]: Can we wrap these in `t.Run()` if we want to add these comments around what the test case is doing?

File: integration/database_sql_mode_test.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
+package integration
```
Comment (zachgersh) [inline integration/database_sql_mode_test.go]: Thank you for adding an integration test! We really appreciate this.

Any feedback around this, please let us know.

**Comment by zachgersh** [comment]
a couple of small things in addition to what @bentranter mentioned. Really appreciate new features.

File: integration/database_sql_mode_test.go (hunk 2 above)
Comment (gregmankes) [reply integration/database_sql_mode_test.go]: got it

File: commands/databases.go (hunk 3 above)
Comment (gregmankes) [reply commands/databases.go]: will do

File: commands/databases_test.go (hunk 4 above)
Comment (gregmankes) [reply commands/databases_test.go]: sure

File: go.mod (hunk 1 above)
Comment (gregmankes) [reply go.mod]: When is that getting merged? If all goes well, this may get in before that :smirk:

File: commands/databases_test.go [hunk 6]
```diff
@@ -858,3 +874,58 @@ func TestDatabasesReplicaDelete(t *testing.T) {
...
```
Comment (zachgersh) [inline commands/databases_test.go]: think we meant to remove this now that we have the `t.Run`

File: commands/databases_test.go [hunk 7]
```diff
@@ -858,3 +874,58 @@ func TestDatabasesReplicaDelete(t *testing.T) {
...
+			tm.databases.EXPECT().GetSQLMode(testDBCluster.ID).Return(testSQLModes, nil)
+
+			config.Args = append(config.Args, testDBCluster.ID)
+
+			err := RunDatabaseGetSQLModes(config)
+			assert.NoError(t, err)
+		})
+	})
+
+	// Error
```
Comment (zachgersh) [inline commands/databases_test.go]: same here, remove please!

File: commands/databases_test.go [hunk 8]
```diff
@@ -858,3 +874,58 @@ func TestDatabasesReplicaDelete(t *testing.T) {
...
+	})
+}
+
+func TestDatabaseSetSQLModes(t *testing.T) {
+	testSQLModesInterface := []interface{}{}
+	for _, sqlMode := range testSQLModes {
+		testSQLModesInterface = append(testSQLModesInterface, sqlMode)
+	}
+
+	// Successful
```
Comment (zachgersh) [inline commands/databases_test.go]: lets add the `t.Runs` here as well and remove the comments.

File: go.sum [hunk 9]
```diff
@@ -32,12 +32,6 @@ github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
 github.com/dgrijalva/jwt-go v0.0.0-20160705203006-01aeca54ebda/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
 github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
 github.com/dgryski/go-sip13 v0.0.0-20181026042036-e10d5fee7954/go.mod h1:vAd38F8PWV+bWy6jNmig1y/TA+kYO4g3RSRF0IAv0no=
-github.com/digitalocean/godo v1.22.0 h1:bVFBKXW2TlynZ9SqmlM6ZSW6UPEzFckltSIUT5NC8L4=
-github.com/digitalocean/godo v1.22.0/go.mod h1:iJnN9rVu6K5LioLxLimlq0uRI+y/eAQjROUmeU/r0hY=
-github.com/digitalocean/godo v1.24.2-0.20191106222530-9ccfbd9c44ff h1:/tA+xwBHCwWdwizotGgjXrosBupethua3SMQ9Rozyrg=
```
Comment (zachgersh) [inline go.sum]: this is interesting that the sum changed? I guess we didn't run a tidy and all this was left over from other pulls?

Comment (zachgersh) [reply go.sum]: cc @bentranter

File: commands/databases_test.go (hunk 8 above)
Comment (gregmankes) [reply commands/databases_test.go]: *scratches head* I remember doing that one too 0.o

File: go.sum (hunk 9 above)
Comment (gregmankes) [reply go.sum]: yeah I just ran a tidy on that one and got this after rebase with master

Comment (zachgersh) [reply go.sum]: @gregmankes - thanks for that. We should have double checked that on the other PR.

**Pull request merged**